vxtelegram/telegram.py

```diff
@@ -157,31 +185,104 @@ def handle_outbound_message(self, message):
         url = self.get_outbound_url('sendMessage')
         http_client = HTTPClient(self.agent_factory())
 
-        try:
-            r = yield http_client.post(
-                url=url,
-                data=json.dumps(outbound_msg),
-                headers={'Content-Type': ['application/json']},
-            )
+        r = yield http_client.post(
+            url=url,
+            data=json.dumps(outbound_msg),
+            headers={'Content-Type': ['application/json']},
+            allow_redirects=False,
+        )
+
         # Telegram redirects our request if our bot token is invalid
-        except ResponseFailed as e:
-            yield self.outbound_failure(message_id,
-                                        'Invalid token (redirected)\n%s' % e)
+        if r.code == http.FOUND:
```

Comment:
@justinvdm so we can assume that the token, even if invalid, is still going to be in the correct format? Because if it's not and we get a 302, then the response body is a mess of HTML and there's no description we can use.